src/server/scriptInfo.ts

```diff
@@ -304,6 +304,7 @@ namespace ts.server {
             const isNew = !this.isAttached(project);
             if (isNew) {
                 this.containingProjects.push(project);
+                project.hasMoreOrLessScriptInfos = true;

```

Comment:
i would rather make this a function call and let the project handle it by setting the flag, instead of setting the flag here.. just an encapsulation comment.